plan/physical_plan_builder.go

```diff
@@ -665,7 +690,23 @@ func (p *Aggregation) handleFinalAgg(x physicalDistSQLPlan, childInfo *physicalP
 	return info
 }
 
-func (p *Aggregation) handleHashAgg() (*physicalPlanInfo, error) {
+// convert2PhysicalPlanCompleteHash converts logical aggregation to complete hash aggregation *physicalPlanInfo.
+func (p *Aggregation) convert2PhysicalPlanCompleteHash(childInfo *physicalPlanInfo) *physicalPlanInfo {
+	agg := &PhysicalAggregation{
+		AggType:      CompleteAgg,
+		AggFuncs:     p.AggFuncs,
+		GroupByItems: p.GroupByItems,
+	}
+	agg.HasGby = len(p.GroupByItems) > 0
+	agg.SetSchema(p.schema)
+	info := addPlanToResponse(agg, childInfo)
+	info.cost += float64(info.count) * memoryFactor
+	info.count = uint64(float64(info.count) * aggFactor)
+	return info
+}
+
+// convert2PhysicalPlanHash converts logical aggregation to physical hash aggregation.
```

Comment:
logical aggregation to physical hash aggregation ->
the logical aggregation to the physical hash aggregation